src/Plugin/Field/FieldType/TypedRelation.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Drupal\controlled_access_terms\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\FormElement;
+use Drupal\Core\TypedData\DataDefinition;
+
+/**
+ * Implements a Typed Relation field.
+ *
+ * @FieldType(
+ *   id = "typed_relation",
+ *   label = @Translation("Typed Relation"),
+ *   module = "controlled_access_terms",
+ *   description = @Translation("Implements a typed relation field"),
+ *   default_formatter = "typed_relation_default",
+ *   default_widget = "typed_relation_default",
+ *   list_class = "\Drupal\Core\Field\EntityReferenceFieldItemList",
+ * )
+ */
+class TypedRelation extends EntityReferenceItem {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function schema(FieldStorageDefinitionInterface $field_definition) {
+    $schema = parent::schema($field_definition);
+    $schema['columns']['rel_type'] = [
+      'type' => 'text',
+      'size' => 'tiny',
+      'not null' => TRUE,
+    ];
```

Comment:
I don't understand what is going on here. Is this adding in the schema? Could/should this be a yaml file?

Reply:
According to the [Drupal 8 "Create a custom field type" documentation](https://www.drupal.org/docs/8/creating-custom-modules/create-a-custom-field-type): "FieldItemInterface::schema() should be overridden, in order to tell the system how to store the value(s) for the field." Every example of custom fields I've seen does it.